app/backend/prepdocslib/blobmanager.py

```diff
@@ -93,7 +131,20 @@ async def upload_pdf_blob_images(
 
         for i in range(page_count):
             blob_name = BlobManager.blob_image_name_from_file_page(file.content.name, i)
-            logger.info("Converting page %s to image and uploading -> %s", i, blob_name)
+
+            blob_client = container_client.get_blob_client(blob_name)
+            if await blob_client.exists():
```

Comment:
Do we need this additional check here, given the check that happens in the function that calls this code?